assets/src/story-embed-block/block/embedControls.js

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import {
+  Button,
+  ToolbarGroup,
+  BaseControl,
+  TextControl,
+  PanelBody,
+  PanelRow,
+} from '@wordpress/components';
+import {
+  BlockControls,
+  InspectorControls,
+  MediaUpload,
+  MediaUploadCheck,
+} from '@wordpress/block-editor';
+import { withInstanceId } from '@wordpress/compose';
+import { createRef, useCallback } from '@wordpress/element';
+
+const POSTER_ALLOWED_MEDIA_TYPES = ['image'];
+
+const EmbedControls = (props) => {
+  const {
+    instanceId,
+    switchBackToURLInput,
+    width,
+    height,
+    minWidth,
+    maxWidth,
+    minHeight,
+    maxHeight,
+    poster,
+    title,
+    setAttributes,
+  } = props;
+
+  const posterDescription = `web-stories-embed-block__poster-image-description-${instanceId}`;
+  const posterImageButton = createRef();
+
+  const onSelectPoster = useCallback(
+    (image) => {
+      setAttributes({ poster: image.url });
+    },
+    [setAttributes]
+  );
+
+  const onRemovePoster = useCallback(() => {
+    setAttributes({ poster: '' });
+
+    // Move focus back to the Media Upload button.
+    posterImageButton.current.focus();
+  }, [setAttributes, posterImageButton]);
+
+  const hasPoster = Boolean(poster);
+
+  return (
+    <>
+      <BlockControls>
+        <ToolbarGroup>
+          <Button
+            className="components-toolbar__control"
+            label={__('Edit URL', 'web-stories')}
+            icon="edit"
+            onClick={switchBackToURLInput}
+          />
+        </ToolbarGroup>
+      </BlockControls>
+      <InspectorControls>
+        <PanelBody title={__('Embed Settings', 'web-stories')}>
+          <PanelRow>
+            <BaseControl>
+              <TextControl
+                type="string"
```

Comment:
```suggestion
                type="text"
```

Reply:
This value is passed to the `<input` type and string is not valid.

Reply:
Good catch! Fixed in 8ab889c342e2f95ecbc0887a167e0a3dd5f16eb7